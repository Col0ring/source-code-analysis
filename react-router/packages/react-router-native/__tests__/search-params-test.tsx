import * as React from "react";
import { View, Text, TextInput } from "react-native";
import {
  NativeRouter,
  Routes,
  Route,
  useSearchParams
} from "react-router-native";
import * as TestRenderer from "react-test-renderer";

describe("useSearchParams", () => {
  function SearchForm({
    children
  }: {
    children: React.ReactNode;
    onSubmit?: any;
  }) {
    return <View>{children}</View>;
  }

  function SearchPage() {
    let [searchParams, setSearchParams] = useSearchParams({ q: "" });
    let [query, setQuery] = React.useState(searchParams.get("q")!);

    function handleSubmit() {
      setSearchParams({ q: query });
    }

    return (
      <View>
        <Text>The current query is "{searchParams.get("q")}".</Text>

        <SearchForm onSubmit={handleSubmit}>
          <TextInput value={query} onChangeText={setQuery} />
        </SearchForm>
      </View>
    );
  }

  it("reads and writes the search string", () => {
    let renderer: TestRenderer.ReactTestRenderer;
    TestRenderer.act(() => {
      renderer = TestRenderer.create(
        <NativeRouter initialEntries={["/search?q=Michael+Jackson"]}>
          <Routes>
            <Route path="search" element={<SearchPage />} />
          </Routes>
        </NativeRouter>
      );
    });

    expect(renderer.toJSON()).toMatchSnapshot();

    let textInput = renderer.root.findByType(TextInput);

    TestRenderer.act(() => {
      textInput.props.onChangeText("Ryan Florence");
    });

    let searchForm = renderer.root.findByType(SearchForm);

    TestRenderer.act(() => {
      searchForm.props.onSubmit();
    });

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
