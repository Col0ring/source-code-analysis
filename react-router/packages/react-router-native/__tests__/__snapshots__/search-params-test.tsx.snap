// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSearchParams reads and writes the search string 1`] = `
<View>
  <Text>
    The current query is "
    Michael Jackson
    ".
  </Text>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      underlineColorAndroid="transparent"
      value="Michael Jackson"
    />
  </View>
</View>
`;

exports[`useSearchParams reads and writes the search string 2`] = `
<View>
  <Text>
    The current query is "
    Ryan Florence
    ".
  </Text>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      underlineColorAndroid="transparent"
      value="Ryan Florence"
    />
  </View>
</View>
`;
